# python errror handling

try:
    print(a)

except NameError :
    print("its a name error")

else:
    print("its all good")

finally:
    print("code fully run")